Userland/Libraries/LibCore/Resource.h

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (c) 2023, Andrew Kaster <akaster@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/ByteBuffer.h>
+#include <AK/Error.h>
+#include <AK/MemoryStream.h>
+#include <AK/OwnPtr.h>
+#include <AK/Span.h>
+#include <AK/String.h>
+#include <AK/StringView.h>
+#include <AK/Variant.h>
+#include <LibCore/File.h>
+#include <LibCore/MappedFile.h>
+
+namespace Core {
+class Resource {
+public:
+    static ErrorOr<NonnullOwnPtr<Resource>> load_from_uri(StringView);
+
+    [[nodiscard]] bool is_file() const { return !m_data.has<DirectoryTag>(); }
+    [[nodiscard]] bool is_directory() const { return m_data.has<DirectoryTag>(); }
+
+    [[nodiscard]] String uri() const;
+    [[nodiscard]] String filename() const;
+    [[nodiscard]] Optional<String> filesystem_path() const;
```

Comment:
LexicalPath probably?

Reply:
The intent of this API is that it will go away once everyone is converted to using resourrces, so I'm not really keen on changing it to represent a lexical path. anyone that really needs a lexical path can get one from it.